File: algorithms/data-structures/arrays/Arrays - DS/arrays-ds.swift
// Arrays - DS
// https://www.hackerrank.com/challenges/arrays-ds/problem

func reverseArray(a: [Int]) -> [Int] {
  var result: [Int] = []
  
  for idx in stride(from: a.count - 1, to: -1, by: -1) {
    result.append(a[idx])
  }
  return result
}

File: algorithms/data-structures/arrays/Arrays - DS/arrays-ds.ts
// Arrays - DS
// https://www.hackerrank.com/challenges/arrays-ds/problem

// function reverseArray(a: number[]): number[] {
//   return a.reverse();
// }

function reverseArray(a: number[]): number[] {
  let result = [];

  for (let i = a.length - 1; i >= 0; i -= 1) {
    result.push(a[i]);
  }
  return result;
}
